### PR TITLE
Add isLastLook, remove canMakerControlSettlement from TakerRequest

### PR DESCRIPTION
**Pull request by phil-ociraptor**
Adding the isLastLook field to TakerRequests in anticipation of RFQM

### CODE REVIEW

File: src/request_parser.ts [hunk 1]
```diff
@@ -62,9 +63,10 @@ export const parseTakerRequest = (req: Pick<express.Request, 'headers' | 'query'
             } else {
                 takerRequest = {
                     ...takerRequestBase,
-                    sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
                     protocolVersion,
+                    sellAmountBaseUnits: new BigNumber(query.sellAmountBaseUnits),
                     txOrigin: query.txOrigin!,
+                    isLastLook,
```
Comment (PirosB3) [inline src/request_parser.ts:69]: Why not add this in `takerRequestBase`

Comment (phil-ociraptor) [reply src/request_parser.ts:69]: As I understand it `isLastLook` is only for v4 - so I only added it to the v4 version. But can add to `takerRequestBase` if needed for v3 and v4

File: src/schemas/taker_request_schema.json [hunk 2]
```diff
@@ -18,6 +18,10 @@
         },
         "txOrigin": {
             "$ref": "/addressSchema"
+        },
+        "isLastLook": {
+            "type": "string",
```
Comment (PirosB3) [inline src/schemas/taker_request_schema.json:23]: Just curious, does the type `bool` or `boolean` work?

File: src/request_parser.ts (hunk 1 above)
Comment (PirosB3) [reply src/request_parser.ts:69]: oh I see it! thanks that makes sense.

Would you be so kind to create a mixin object that we can use to merge v4-related taker parameters without having to duplicate them?

Comment (phil-ociraptor) [reply src/request_parser.ts]: Done!

**Review by PirosB3 — APPROVED** [review]

**Review by PirosB3 — APPROVED** [review]

**Pull request merged**